### ISSUE TITLE
Add informational banner to entries that are linked with KeeShare

### ISSUE THREAD

**Issue by rocketraman** [keepassxreboot/keepassxc#3800]
## Summary
When viewing Keeshared entries from another database, modifications to those entries are "useless" as those entries are just a view of the entries that are actually in another database. It is easy to search for an entry and modify it, not realizing it comes from another database and should actually be modified there.

## Desired Behavior
Just make Keeshared entries read-only so they can't be modified. Or alternatively, support modifying them by writing back to the source db.

**Comment by droidmonkey** [comment]
This is only true if you are importing only

**Comment by rocketraman** [comment]
I (think) I am importing only. Because everything gets imported with no structure, I decided that was best for now.

**Comment by rocketraman** [comment]
Yes, Type=Import. I do note that the imported entries do have structure now. However, I was sure when I tried "Synchronize" before things got messed up. Perhaps it is time to try Sync again, but in any case, if Type=Import, the entries should be read-only.

**Comment by rocketraman** [comment]
Yeah, I think I set it to Import-only for now due to problems similar to https://github.com/keepassxreboot/keepassxc/issues/3045.

**Comment by dssouza-ti** [comment]
@rocketraman I've been using the #3045 code to sync group structure and it's working really nice for me. If this is something that would be useful to you, please test it and report to @dsonck92 on the aforementioned issue.

**Comment by droidmonkey** [comment]
I don't think we should make the entry read only, there are several good reasons why you would want to make changes to an imported entry. The merge operation preserves those changes. We should display a warning banner for entries that are synced/exported/imported using KeeShare though.

**Comment by droidmonkey** [comment]
Will be obsolete with the major changes to KeeShare (making it group-bound) so this will not be implemented.

**Comment by rocketraman** [comment]
> Will be obsolete with the major changes to KeeShare (making it group-bound) so this will not be implemented.

I'm assuming the UI for the group-bound KeeShare makes it very clear which groups are KeeShare groups? And flows that down to the entries somehow?

**Comment by droidmonkey** [comment]
Yes it will be obvious (already is) when a group is shared.

**Comment by rocketraman** [comment]
Having now used 2.7.1 for a day or so, I still think this issue should be considered for re-opening and implementation.

At the group level it is now much clearer what is shared and what isn't, however, when a keeshared entry is double-clicked, there remains no more indication whatsoever on the screen that the entry comes from a keeshared group. In fact, even the name of the group which is bound via Keeshare is not shown on the screen (only sub-groups of that group are, for some reason), so even that "hint" is unavailable.

The banner at the top of the password list view "Imported from /foo/bar/baz.kdbx" is useful -- why not show the same banner at the entry level as well?

**Comment by rocketraman** [comment]
Thanks for re-opening... thought of another situation. If one clicks on an entry from search results, then a keeshared entry can be opened *without* even seeing the "Imported from /foo/bar/baz.kdbx" banner in the password list view first, making it even more critical than some kind of banner at the entry level is shown.

**Comment by droidmonkey** [comment]
I mean to be fair, you choose to incorporate keeshare and you should be aware of which groups are shared... groups are shown on search.

**Comment by rocketraman** [comment]
> I mean to be fair, you choose to incorporate keeshare and you should be aware of which groups are shared... groups are shown on search.

They are, but if a group is imported with structure, then the "Group" column is only the last level in the structure shown, which isn't necessarily unique to the Keeshared groups (and isn't in my databases). The entire group path can still be seen in the details view at the bottom when an entry is selected, but that is relatively easy to miss.

**Comment by santo74** [comment]
I would like to add that not necessarily all users are "technical" people, meaning that they don't necessarily have a clue what the message "imported from..." means, let alone that they understand what it implies.
They might even not pay attention to such message at all.

We for example have several databases that contain passwords for specific use cases and those passwords need to be accessible for some people, depending on their role.
This is done by giving everyone a personal database (where they can manage their own, personal passwords) and then importing one or more of the specific databases with KeeShare.
This way, they have access to all the passwords they need.

If a new password needs to be shared or an existing modified, this should obviously be done in the original database that's being shared. However, since they don't have a clue what "imported" means they sometimes modify an item in the imported group, thinking that everyone else with access to the shared database will see those changes.
Even worse, they sometimes add a new entry somewhere in the imported group, e.g. when they registered a new account that should be available for the others as well.

The worst thing here is that they don't get any errors or warnings when doing so and therefore they are not aware that their modifications / additions are not visible for the others.

So it would be really nice if an import could be made read-only.
And since I understand there are also good reasons to be able to make changes to imported entries, I suggest to implement it as an option (in the KeeShare options for a group)
